content/nginx/deployment-guides/single-sign-on/active-directory-federation-services.md

```diff
@@ -1,153 +1,328 @@
+       "issuer": "https://adfs-server-address/adfs",
+       "authorization_endpoint": "https://adfs-server-address/adfs/oauth2/authorize/",
+       "token_endpoint": "https://adfs-server-address/adfs/oauth2/token/",
+       "jwks_uri": "https://adfs-server-address/adfs/discovery/keys",
        ...
-       "token_endpoint": "https://<ADFS-server-address>/oidc/adfs/token",
-    ...
    }
    ```
 
-   <span id="nginx-plus-variables"></span>
-4. In your preferred text editor, open **/etc/nginx/conf.d/frontend.conf**. Change the "default" parameter value of each of the following [map](https://nginx.org/en/docs/http/ngx_http_map_module.html#map) directives to the specified value:
+   <span id="adfs-setup-issuer"></span>
+2. Copy the **issuer** value, you will need it later when configuring NGINX Plus. Typically, the OpenID Connect Issuer for AD FS is:
+
+   `https://adfs-server-address/adfs`.
+
+
+{{< note >}} You will need the values of **Client ID**, **Client Secret**, and **Issuer** in the next steps. {{< /note >}}
+
+
+## Set up NGINX Plus {#nginx-plus-setup}
+
+With AF DS configured, you can enable OIDC on NGINX Plus. NGINX Plus serves as the Rely Party (RP) application &mdash; a client service that verifies user identity.
+
+
+1.  Ensure that you are using the latest version of NGINX Plus by running the `nginx -v` command in a terminal:
+
+    ```shell
+    nginx -v
+    ```
+    The output should match NGINX Plus Release 34 or later:
+
+    ```none
+    nginx version: nginx/1.27.4 (nginx-plus-r34)
+    ```
+
+2.  Ensure that you have the values of the **Client ID**, **Client Secret**, and **Issuer** obtained during [AD FS Configuration](#adfs-setup).
+
+3.  In your preferred text editor, open the NGINX configuration file (`/etc/nginx/nginx.conf` for Linux or `/usr/local/etc/nginx/nginx.conf` for FreeBSD).
+
+4.  In the [`http {}`](https://nginx.org/en/docs/http/ngx_http_core_module.html#http) context, make sure your public DNS resolver is specified with the [`resolver`](https://nginx.org/en/docs/http/ngx_http_core_module.html#resolver) directive. By default, NGINX Plus re‑resolves DNS records at the frequency specified by time‑to‑live (TTL) in the record, but you can override the TTL value with the `valid` parameter:
+
+    ```nginx
+    http {
+        resolver 10.0.0.1 ipv4=on valid=300s;
+
+        # ...
+    }
+    ```
+    <span id="adfs-setup-oidc-provider"></span>
+5.  In the [`http {}`](https://nginx.org/en/docs/http/ngx_http_core_module.html#http) context, define the AD FS OIDC provider named `adfs` by specifying the [`oidc_provider {}`](https://nginx.org/en/docs/http/ngx_http_oidc_module.html#oidc_provider) context:
+
+    ```nginx
+    http {
+        resolver 10.0.0.1 ipv4=on valid=300s;
+
+        oidc_provider adfs {
+
+            # ...
+
+        }
+        # ...
+    }
+    ```
+
+6.  In the [`oidc_provider {}`](https://nginx.org/en/docs/http/ngx_http_oidc_module.html#oidc_provider) context, specify:
```

Comment:
All of these URLs 404: 
https://nginx.org/en/docs/http/ngx_http_oidc_module.html